src/Authoring/WinRT.SourceGenerator/Generator.cs

```diff
@@ -28,91 +31,46 @@ internal sealed class WriteOnlyArrayAttribute : global::System.Attribute
-
-        private static string GetCsWinRTExe(GeneratorExecutionContext context)
-        {
-            context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.CsWinRTExe", out var cswinrtExe);
-            return cswinrtExe;
-        }
-
-        private static bool GetKeepGeneratedSources(GeneratorExecutionContext context)
-        {
-            context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.CsWinRTKeepGeneratedSources", out var keepGeneratedSourcesStr);
-            return keepGeneratedSourcesStr != null && bool.TryParse(keepGeneratedSourcesStr, out var keepGeneratedSources) && keepGeneratedSources;
-        }
-
-        private static string GetCsWinRTWindowsMetadata(GeneratorExecutionContext context)
-        {
-            context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.CsWinRTWindowsMetadata", out var cswinrtWindowsMetadata);
-            return cswinrtWindowsMetadata;
-        }
-
-        private static string GetCsWinRTDependentMetadata(GeneratorExecutionContext context)
-        {
-            context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.CsWinRTAuthoringInputs", out var winmds);
-            return winmds;
+        public ComponentGenerator(GeneratorExecutionContext context)
+        {
+            this.context = context;
+            Logger = new Logger(context);
         }
 
         private string GetTempFolder(bool clearSourceFilesFromFolder = false)
         {
-            if(_tempFolder == null || !File.Exists(_tempFolder))
+            if (string.IsNullOrEmpty(tempFolder) || !File.Exists(tempFolder))
             {
                 string outputDir = Path.Combine(Path.GetTempPath(), "CsWinRT", Path.GetRandomFileName()).TrimEnd('\\');
                 Directory.CreateDirectory(outputDir);
-                _tempFolder = outputDir;
-                Logger.Log("Created temp folder: " + _tempFolder);
+                tempFolder = outputDir;
+                Logger.Log("Created temp folder: " + tempFolder);
             }
 
             if (clearSourceFilesFromFolder)
             {
-                foreach (var file in Directory.GetFiles(_tempFolder, "*.cs", SearchOption.TopDirectoryOnly))
+                foreach (var file in Directory.GetFiles(tempFolder, "*.cs", SearchOption.TopDirectoryOnly))
                 {
                     Logger.Log("Clearing " + file);
                     File.Delete(file);
                 }
             }
 
-            return _tempFolder;
+            return tempFolder;
         }
 
-        private void GenerateSources(GeneratorExecutionContext context)
+        private void GenerateSources()
         {
-            string cswinrtExe = GetCsWinRTExe(context);
-            string assemblyName = GetAssemblyName(context);
-            string winmdFile = GetWinmdOutputFile(context);
+            string cswinrtExe = context.GetCsWinRTExe();
+            string assemblyName = context.GetAssemblyName();
```

Comment:
Is GetAssemblyName doing the same thing as we were doing? I.e. pulling form analyzerconfigoptions?